Backport CUMULUS-2955 (#2982) to v12

Jkovarik opened the pull request:
CUMULUS-2955 -- Modify migrations to remove update to add files index (#2982)

* Modify migrations to remove update to add files index

* update CHANGELOG

* Include missing/mangled commit

* Minor update**Summary:** Summary of changes

Addresses [CUMULUS-2955](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2955)


## PR Checklist

- [ ] Update CHANGELOG
- [ ] Unit tests
- [ ] Ad-hoc testing - Deploy changes and test manually
- [ ] Integration tests

npauzenga reviewed (changes requested):
Looks good @Jkovarik. It pulls in the correct migration and the version bumps look fine. I added a suggestion for the changelog. Happy to discuss.

Jkovarik commented on CHANGELOG.md:
```diff
@@ -4,10 +4,52 @@ All notable changes to this project will be documented in this file.
+All migration instructions from v12.0.0 should be followed in addition to the following if
+upgrading from a version prior to version 12.
+
+- The changes introduced in CUMULUS-2955 should result in removal of
```
Great plan!   Have a changelog entry! :thumbsup:  ff12db5cba

Jkovarik commented:
I *think* this has addressed the feedback @npauzenga  however..... it's an open question if we want to bother with this backport given the state of #3023/CUMULUS-2692
